Mage.Sets/src/mage/cards/d/DonalHeraldOfWings.java

```diff
@@ -0,0 +1,145 @@
+    static {
+        filterSpell.add(Predicates.not(SuperType.LEGENDARY.getPredicate()));
+        filterSpell.add(CardType.CREATURE.getPredicate());
+    }
+
+    public DonalHeraldOfWings(UUID ownderId, CardSetInfo setInfo) {
+        super(ownderId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}{U}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+
+        this.addSubType(SubType.HUMAN);
+        this.addSubType(SubType.WIZARD);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Whenever you cast a nonlegendary creature spell with flying, you may copy it,
+        // except the copy is a 1/1 Spirit in addition to its other types.
+        // Do this only once each turn. (The copy becomes a token.)
+        // TODO: This still triggers and asks if you wanna use it, even if you've used it once this turn.
+        Ability ability = new SpellCastControllerTriggeredAbility(new DonalHeraldOfWingsEffect(), filterSpell, true, true);
+        ability.setIdentifier(MageIdentifier.DonalHeraldOfWingsWatcher);
+        this.addAbility(ability, new DonalHeraldOfWingsWatcher());
+    }
+
+    private DonalHeraldOfWings(final DonalHeraldOfWings card) { super(card); }
+
+    @Override
+    public DonalHeraldOfWings copy() { return new DonalHeraldOfWings(this); }
+}
+
+class DonalHeraldOfWingsEffect extends OneShotEffect {
+
+    DonalHeraldOfWingsEffect() {
+        super(Outcome.Copy);
+        staticText = "you may copy it, except the copy is a 1/1 Spirit in addition to its other types.";
+    }
+
+    private DonalHeraldOfWingsEffect(final DonalHeraldOfWingsEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) { return false; }
+
+        DonalHeraldOfWingsWatcher watcher = game.getState().getWatcher(DonalHeraldOfWingsWatcher.class);
+        if (watcher == null) { return false; }
+        if (watcher.wasAbilityUsedThisTurn()) { return false; }
+
+        // Get the card that was cast
+        if (this.getTargetPointer() == null) { return false; }
+        Spell originalSpell = game.getStack().getSpell(((FixedTarget) this.getTargetPointer()).getTarget());
+
+        // Create a token copy
+        // TODO: Copy made is a token, but there's no indication of it on the card
+        originalSpell.createCopyOnStack(game, source, controller.getId(), false, 1, DonalHeraldOfWingsApplier.instance);
+
+        return true;
+    }
+
+    @Override
+    public Effect copy() { return new DonalHeraldOfWingsEffect(this); }
+}
+
+enum DonalHeraldOfWingsApplier implements StackObjectCopyApplier {
+    instance;
+
+    @Override
+    public void modifySpell(StackObject copiedSpell, Game game) {
+        copiedSpell.addSubType(SubType.SPIRIT);
+        // TODO: These two calls don't work
```

Comment:
The creature gains the spirit ability, but the power and toughness does not seem to get changed

Reply:
try using `modifyBaseValue`